### PR TITLE
Add in Webpack to parse multiple front end modules

### PR DESCRIPTION
**Pull request by lauraGgit**
- refactor to a webpack compile and add in an eslint to better support modular development
- see [readme](https://github.com/18F/analytics.usa.gov/tree/webpack-revert-index-js#javascript-modules) for updates on the module structure.

- saves an estimated 1mb in unneeded js!

# Tasks
- [X] install wepback
- [x] break into modules
- [x] dry up some of the barcharts and percent calculations
- [x] fix `formatCommas`
- [x] fix number of browsers/os shown (in helpers.collaspe others)
- [x] decide whether to `.gitignore` bundle.js
- [x] bump package version to `2.0.0`
- [x] add webpack scripts into ci
- [x] add local watch mode
- [x] further restrict eslint rules
- [x] move from vendor scripts to npm imports
- [x] update readme

### CODE REVIEW

**Comment by lauraGgit** [comment]
This is ready but for documentation updates.

going to move the git hooks into a separate pr.

**Comment by tdlowden** [comment]
cc @konklone

File: README.md [hunk 1]
```diff
@@ -31,30 +31,39 @@ Other organizations who have reused this project for their analytics dashboard:
 
 [This blog post details their implementations and lessons learned](https://18f.gsa.gov/2016/01/05/tips-for-adapting-analytics-usa-gov/).
 
+## About the components
+Ths app uses [Jekyll](http://jekyllrb.com) to build the site, and [Sass](http://sass-lang.com/), [Bourbon](http://bourbon.io), and [Neat](http://neat.bourbon.io) for CSS.
```
Comment (konklone) [inline README.md]: FYI, these sites support `https://` now. =)

File: _includes/footer_scripts.html [hunk 2]
```diff
@@ -17,5 +17,4 @@
     window._ie = window._ie9 || window._ie10 || window._ie11;
   </script>
 
-  <script src="{{ site.baseurl }}/js/vendor/uswds.v0.9.6.js"></script>
-  <script src="{{ site.baseurl }}/js/index.js"></script>
+  <script src="{{ site.baseurl }}/assets/bundle.js"></script>
```
Comment (konklone) [inline _includes/footer_scripts.html]: Since we support HTTP/2 (which supports multiplexing), what's the advantage of packing these assets into one file?

**Comment by tdlowden** [comment]
@lauraGgit are we good to go?

**Pull request merged**